ttnn/cpp/ttnn/decorators.hpp

```diff
@@ -203,51 +210,72 @@ struct registered_operation_t {
         return detail::python_fully_qualified_name(std::string{cpp_fully_qualified_name});
     }
 
-    template <typename... args_t>
-    requires PrimitiveOperationConcept<operation_t>
-    auto invoke(uint8_t queue_id, args_t&&... args) const {
-        static_assert(requires { operation_t::invoke(std::forward<decltype(args)>(args)...); },
-                      "Primitive Operation must implement operator() method to be invoked.");
+    auto operator()(uint8_t queue_id, const tensor_args_t& tensor_args, const operation_attributes_t& operation_attributes = {}) const {
```

Comment:
Can we _please_ expand these as the individual elements of the tuples like in https://godbolt.org/z/cWdTPnsrh?